exercises.ipynb

```diff
@@ -40,71 +38,63 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 273,
-   "metadata": {
-    "autoscroll": false,
-    "collapsed": false,
-    "ein.hycell": false,
-    "ein.tags": "worksheet-0",
-    "slideshow": {
-     "slide_type": "-"
-    }
-   },
+   "execution_count": 3,
+   "metadata": {},
    "outputs": [],
    "source": [
     "def get_titles(soup):    \n",
     "    \"\"\" Returns a list of titles on the page \"\"\"\n",
-    "    # the \"soup\" parameter is of the type that is\n",
-    "    # returned by Beautiful Soup when it parses HTML.\n",
-    "    # The function should then use the object to\n",
-    "    # extract a list of titles (of the lego sets)\n",
-    "    #\n",
-    "    # Lookup the documentation for Beautiful Soup\n",
-    "    # Figure out how to select the text of the title\n",
-    "    # of each legoset. A title should look like: \n",
-    "    # \"10252: Volkswagen Beetle\"\n",
-    "    pass\n",
+    "    titles = [x.find_all(\"a\")[0].get_text() for x in soup.find_all(class_=[\"meta\",\"h1\"])]\n",
+    "    return titles\n",
     "\n",
     "def parse_bricks(url):\n",
     "    \"\"\" Fetches Lego Bricks page and extracts titles \"\"\"\n",
-    "    # Lookup the documentation to the \"requests\" library\n",
-    "    #\n",
-    "    # Use requests to make a get request to the\n",
-    "    # url given in the argument \"url\" (which is a string)\n",
-    "    # and get the raw HTML body of the response\n",
-    "    #\n",
-    "    # Use \"BeautifulSoup\" to parse this HTML. \n",
-    "    #\n",
-    "    # Use the \"get_titles\" function to extract the\n",
-    "    # titles from the BeautifulSoup object.\n",
-    "    #\n",
-    "    # Return the titles\n",
-    "    pass"
+    "    titles = get_titles(BeautifulSoup(requests.get(START_URL).content))\n",
+    "    return titles"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 276,
-   "metadata": {
-    "autoscroll": false,
-    "collapsed": false,
-    "ein.hycell": false,
-    "ein.tags": "worksheet-0",
-    "slideshow": {
-     "slide_type": "-"
-    }
-   },
+   "execution_count": 4,
+   "metadata": {},
    "outputs": [],
    "source": [
-    "bricks = parse_bricks(START_URL)"
+    "bricks = []\n",
+    "next_url = START_URL\n",
+    "while next_url != 0:\n",
+    "    soup = BeautifulSoup(requests.get(next_url).content)\n",
+    "    for i in get_titles(soup):\n",
+    "        bricks.append(i)\n",
+    "    try: \n",
+    "        next_url = soup.select(\"li.next a\")[0][\"href\"]\n",
+    "    except:\n",
```

Comment:
You can just break the while loop directly! Or set next_url to False... more Pythonic! Otherwise, looks good!